### PR TITLE
Update to stable futures

### LINKED ISSUE #50: Update to futures 0.3
Blocked on https://github.com/hyperium/hyper/pull/2007 and on `soketto` updating.

### PR DESCRIPTION
**Pull request by ascjones**
Closes #50.

- [x] fix tests:`no current reactor` at https://github.com/paritytech/jsonrpsee/blob/1a5a8d132daad84fc9b2f0c003b7c0ce41c9cc7d/http/src/server/background.rs#L77

### CODE REVIEW

**Comment by tomaka** [comment]
> fix tests:no current reactor at 

You have to update to tokio v0.2.

**Comment by ascjones** [comment]
Same failure still

**Comment by ascjones** [comment]
I think this is why: 

https://github.com/hyperium/hyper/commit/cb3f39c2dc6340060f6b17f354f04c872a947574#diff-afb1712a59f738bc3dfe39004cb5e4b9L132

Previously using `Handle::default()` ([lazy](https://tokio.rs/docs/internals/net/#handle-current-vs-handle-default)) where now it goes to `Handle::current()` (eager) as per https://github.com/tokio-rs/tokio/pull/1666. 

So not sure how to fix this - just discovered it so will try a few things. Posting here in case you have any ideas @tomaka

File: http/src/server/background.rs [hunk 1]
```diff
@@ -93,13 +93,27 @@ impl BackgroundHttp {
                 };
 
                 runtime.block_on(async move {
+                    let builder = match hyper::Server::try_bind(&addr) {
```
Comment (tomaka) [inline http/src/server/background.rs]: We reaaaaally want `bind` to fail if `hyper::Server::try_bind` fails. That's the whole point of returning an error in `bind`.

Comment (ascjones) [reply http/src/server/background.rs]: :man_facepalming:

File: http/src/server/background.rs [hunk 2]
```diff
@@ -93,13 +99,25 @@ impl BackgroundHttp {
...
+                        }
+                        Err(err) => {
+                            log::error!("Failed to bind to address {}: {}", addr, err);
+                            let _ = addr_tx.send(Err(err));
+                            return;
+                        }
+                    };
                 });
             })
             .unwrap();
```
Comment (tomaka) [inline http/src/server/background.rs]: While you're at it:

```suggestion
            ?;
```

File: ws/src/client.rs [hunk 3]
```diff
@@ -246,10 +245,10 @@ impl<'a> WsTransportClientBuilder<'a> {
             }
         }
 
+        let (sender, receiver) = client.into_builder().finish();
+
         // If the handshake succeeded, return.
```
Comment (tomaka) [inline ws/src/client.rs]:
```suggestion
        // If the handshake succeeded, return.
        let (sender, receiver) = client.into_builder().finish();
```

File: ws/src/client.rs [hunk 4]
```diff
@@ -246,10 +245,10 @@ impl<'a> WsTransportClientBuilder<'a> {
             }
         }
 
+        let (sender, receiver) = client.into_builder().finish();
```
Comment (tomaka) [inline ws/src/client.rs]:
```suggestion
```

File: ws/src/client.rs [hunk 5]
```diff
@@ -246,10 +245,10 @@ impl<'a> WsTransportClientBuilder<'a> {
...
+
```
Comment (tomaka) [inline ws/src/client.rs]:
```suggestion
```

**Pull request merged**